src/Utility.ts

```diff
@@ -87,63 +66,48 @@ export namespace Utility {
                 return resolve(port);
             });
             server.on('error', (err: Error) => {
-                return reject(new Error(err.toString()));
+                return reject(err);
             });
             server.listen(0, '127.0.0.1');
         });
     }
 
-    export async function openFileIfExists(filepath: string): Promise<boolean> {
-        const exists: boolean = await fse.pathExists(filepath);
-        if (exists) {
-            await vscode.window.showTextDocument(vscode.Uri.file(filepath), { preview: false });
-            return true;
-        } else {
-            return false;
-        }
-    }
-
     export async function getServerPort(serverXml: string): Promise<string> | undefined {
         if (!await fse.pathExists(serverXml)) {
             throw new Error(localize('tomcatExt.noserver', 'No tomcat server.'));
         }
         const xml: string = await fse.readFile(serverXml, 'utf8');
         const jsonObj: {} = await parseXml(xml);
```

Comment:
I'm thinking if we can define an interface for the xml parsed object, then no object access via string literals any more.

Reply:
Considering it too. But in this server port scenario, the schema is a little special. It not only about get object with name, but to filter with some conditions.